### PR TITLE
Add constructor for pre-existing DynamicWatcher

### PR DESCRIPTION
**Pull request by JustinKuli**
This allows users that already have a DynamicWatcher and are using Query Batches to supply their instance to the template resolver. Most details of managing the watcher and the batch are delegated to the user: this package will not automatically stop the query batch after the template is resolved, for instance.

### CODE REVIEW

**Comment by JustinKuli** [comment]
/hold

I believe this is ready for review, but I want to run and end-to-end kind of thing with it before it merges.

File: pkg/templates/templates.go
Comment (mprahl) [inline pkg/templates/templates.go:274]: Could you please document that batch management is disabled?

**Comment by JustinKuli** [comment]
https://github.com/open-cluster-management-io/config-policy-controller/pull/235 is an example use of this (also using another change to this package which I think would be a good separate PR).

**Review by mprahl — APPROVED** [review]

**Pull request merged**
